bindings/wasm/src/error.rs

```diff
@@ -90,9 +91,77 @@ impl_wasm_error_from!(
   identity::credential::Error,
   identity::did::Error,
   identity::did::DIDError,
-  identity::iota::Error
+  identity::iota::ValidationError
 );
 
+// Similar to `impl_wasm_error_from`, but uses the types name instead of requiring/calling Into &'static str
+#[macro_export]
+macro_rules! impl_wasm_error_from_with_struct_name {
+  ( $($t:ty),* ) => {
+  $(impl From<$t> for WasmError<'_> {
+    fn from(error: $t) -> Self {
+      Self {
+        message: Cow::Owned(error.to_string()),
+        name: Cow::Borrowed(stringify!($t)),
+      }
+    }
+  })*
+  }
+}
+
+// identity::iota now has some errors where the error message does not include the source error's error message.
+// This is in compliance with the Rust error handling project group's recommendation:
+// * An error type with a source error should either return that error via source or include that source's error message
+//   in its own Display output, but never both. *
+// See https://blog.rust-lang.org/inside-rust/2021/07/01/What-the-error-handling-project-group-is-working-towards.html#guidelines-for-implementing-displayfmt-and-errorsource.
+//
+// However in WasmError we want the display message of the entire error chain. We introduce a workaround here that let's
+// us display the entire display chain for new variants that don't include the error message of the source error in its
+// own display.
+
+// the following function is inspired by https://www.lpalmieri.com/posts/error-handling-rust/#error-source
+fn error_chain_fmt(e: &impl std::error::Error, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+  write!(f, "{}. ", e)?;
+  let mut current = e.source();
+  while let Some(cause) = current {
+    write!(f, "Caused by: {}. ", cause)?;
+    current = cause.source();
+  }
+  Ok(())
+}
+
+struct ErrorMessage<'a, E: std::error::Error>(&'a E);
+
+impl<'a> Display for ErrorMessage<'a, identity::iota::Error> {
+  fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+    match &self.0 {
+      identity::iota::Error::CredentialValidationError(e) => {
+        write!(f, "{}. ", self.0)?;
+        error_chain_fmt(&e, f)
+      }
+      identity::iota::Error::PresentationValidationError(e) => {
+        write!(f, "{}. ", self.0)?;
+        error_chain_fmt(&e, f)
+      }
+      identity::iota::Error::IsolatedValidationError(e) => {
+        write!(f, "{}. ", self.0)?;
+        error_chain_fmt(&e, f)
+      }
+      // the rest include the source error's message in their own
+      _ => self.0.fmt(f),
+    }
+  }
+}
```

Comment:
As mentioned in a previous review, we need to find a way to generalise the error chain formatting to all errors in Wasm, maybe implement it for any `std::error::Error`. Not in this PR but I would appreciate noting it somewhere or in an issue.

Reply:
I would like to use `anyhow` in the bindings as you suggested earlier. We just need to make sure we don't duplicate error messages first. Other suggestions are of course also welcome.